### ISSUE TITLE
Auto-update not working

### ISSUE THREAD

**Issue by bajtos** [filecoin-station/filecoin-station#284]
My Station installation has been repeatedly not able to auto-upgrade to the next version (0.10.0 -> 0.10.1, 0.10.1 -> 0.10.2, 0.10.2 -> 0.11.0).

The auto-updater module detects a new version, downloads the installer and asks me to restart the app (shows a top banner in the UI). When I click on "Restart Now", the old version of Station starts again.

**Workaround**

1. Download the new version from GitHub releases (or https://filstation.io)
2. Run the installer
3. Tell macOS to `Replace` the current version with the new one

<img width="516" alt="Screenshot 2022-11-23 at 15 07 29" src="https://user-images.githubusercontent.com/1140553/203567144-bfa9bd9f-0499-4c83-85a4-6e5c58059d9f.png">

**Comment by bajtos** [comment]
After upgrading manually, Station reports the following build version:

```
Filecoin Station build version: 0.11.0.1-dev
```

I think this is the same problem that was breaking Sentry error reporting, see #263

**Comment by juliangruber** [comment]
> When I started the app from terminal, I noticed a wrong build version is reported - maybe that's the problem?

Iiuc the auto updater only reads `.version` from `package.json`, which doesn't have the `-dev` suffix, right?

**Comment by bajtos** [comment]
> > When I started the app from terminal, I noticed a wrong build version is reported - maybe that's the problem?
> 
> Iiuc the auto updater only reads `.version` from `package.json`, which doesn't have the `-dev` suffix, right?

I am not sure what information is auto-updater using.

You are right that the `.version` field in `package.json` does not have `-dev` suffix.

**Comment by bajtos** [comment]
Possibly related:
- https://github.com/electron-userland/electron-builder/issues/7121
- https://github.com/electron-userland/electron-builder/issues/6699
-

**Comment by bajtos** [comment]
Since we know electron-autoupdater is broken on macOS Ventura, there is no clear timeline for the issue to get fixed, and we don't have the expertise to fix it ourselves, then maybe we should implement a temporary measure to help our users and prevent confusion.

When there is a new version available, and we detect macOS Ventura, and the user clicks on "restart to update", then we can warn them that this operation may not work and point them to this GH issue.

A better idea: we can detect the situation when "restart-to-update" does not update the version, e.g. by storing a flag in a config file before restarting, and then tell the user about the problem & explain how to upgrade manually. I think this would be the best UX, and it's also something we can keep enabled even after Electron fixes the auto upgrade on macOS Ventura.

Step-by-step proposal:
1. Before calling "restart to update", create a new config file storing the current version
2. At startup, check if that config file is present.
   -  If the file is present and our current version is the same as the one in the config file, we have experienced an auto upgrade issue. Delete the config file. Explain the problem to the user and describe a workaround.
   - If the file is present and our current version is different from the one in the config file, then all is good, and we can delete the config file.
   - If the file is not present, then we are done.

WDYT?

**Comment by juliangruber** [comment]
> Step-by-step proposal:

This sounds really good to me! Stays out of the way and fixes the issue generically

**Comment by momack2** [comment]
re #428 - I'm on MacOS Monterey, not Ventura.

IPFS Desktop updates do work, Station does not.

**Comment by juliangruber** [comment]
I found an issue inside the shipit logs:

```
$ tail ~/Library/Caches/io.filecoin.station.ShipIt/ShipIt_stderr.log
...
2023-02-09 23:30:22.646 ShipIt[37145:5709636] Installation error: Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied" UserInfo={NSURL=file:///private/var/folders/ph/cz_m_sjn3ys1cnnsmx5058tw0000gn/T/io.filecoin.station.ShipIt.TYakqGgm/Filecoin%20Station.app/Contents/Resources/saturn-l2-node/saturn-L2-node, NSLocalizedDescription=Permission denied, NSLocalizedFailureReason=Couldn't remove quarantine attribute from "/private/var/folders/ph/cz_m_sjn3ys1cnnsmx5058tw0000gn/T/io.filecoin.station.ShipIt.TYakqGgm/Filecoin Station.app/Contents/Resources/saturn-l2-node/saturn-L2-node". This most likely means the file is read-only.}
...
```

It looks like the `saturn-L2-node` binary prevents the update. I'll try updating its file permissions

**Comment by juliangruber** [comment]
Initial fix in https://github.com/filecoin-station/filecoin-station/commit/ea6fe8193a2ddc05053606b2da5e1039f04b1e91, whether this unbreaks updates needs to be confirmed.

**Comment by juliangruber** [comment]
This should be fixed in 0.13.1.

I have just verified that auto update from 0.12.1 -> 0.13.1 works.

Please re-open if there are still issues.
